### PR TITLE
Update Provider.react.js

### PR DESCRIPTION
**Pull request by cfloresdfw**
Fetch the labels from our internal API

### CODE REVIEW

File: Provider.react.js [hunk 1]
```diff
@@ -1,6 +1,47 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 import LocaleContext from './Context';
+import {getHostName} from 'virtual-tour-viewer/src/utils/UIHelper';
+
+async function isLocalHost() {
+    let hostname = !window.origin ? window.location.origin : window.origin;
+    return hostname.includes('localhost');
```
Comment (dipeenverma1609) [inline Provider.react.js:8]: create 'localhost' as a const

File: Provider.react.js [hunk 2]
```diff
@@ -1,6 +1,47 @@
...
+}
+
+async function getLabelsFromHavrApi(locale) {
```
Comment (dipeenverma1609) [inline Provider.react.js:11]: correct func name, I can't understand HavrApi

File: Provider.react.js [hunk 3]
```diff
@@ -1,6 +1,47 @@
...
+    let taxonomyName = 'spaces';
+    // defaulting to en-US in case locale isn't present
+    locale = locale ? locale : 'en-US';
```
Comment (dipeenverma1609) [inline Provider.react.js:14]: Create a const DEFAULT_LOCALE = 'en-US'

File: Provider.react.js [hunk 4]
```diff
@@ -1,6 +1,47 @@
...
+    let tardisResponse;
+    if (await isLocalHost()) {
+        tardisResponse = await fetch(
+            `http://localhost:8080/havr-api/v2/taxonomies/${taxonomyName}/${locale}`
```
Comment (dipeenverma1609) [inline Provider.react.js:18]: Do not hardcode, use getHostName() to get host and append rest of the endpoint path.

File: Provider.react.js [hunk 5]
```diff
@@ -1,6 +1,47 @@
...
+        );
+        // if the viewer is running locally without havr-api, hit svc-stage instead
+        if (!tardisResponse.status !== '200') {
+            tardisResponse = await fetch(
+                `https://svc-stage.homeaway.com/havr-api/v2/taxonomies/${taxonomyName}/${locale}`
```
Comment (dipeenverma1609) [inline Provider.react.js:23]: Do not hardcode, use getHostName() to get host and append rest of the endpoint path.

File: Provider.react.js [hunk 6]
```diff
@@ -1,6 +1,47 @@
...
+            );
+        }
+    } else {
+        tardisResponse = await fetch(`${getHostName()}/v2/taxonomies/${taxonomyName}/${locale}`);
+    }
+    return await tardisResponse.json();
+}
+
+async function getTranslations(locale) {
```
Comment (dipeenverma1609) [inline Provider.react.js:32]: add unit tests for the func

File: Provider.react.js (hunk 2 above)
Comment (dipeenverma1609) [inline Provider.react.js:11]: add unit tests for the func

File: Provider.react.js [hunk 7]
```diff
@@ -1,6 +1,47 @@
...
+    let response = await fetch(`static_assets/translations/${locale}.json`);
+    if (response.status !== 200) {
```
Comment (dipeenverma1609) [inline Provider.react.js:34]: Please explain why we are checking for 200 status ?

**Comment by cfloresdfw** [comment]
closing

**Pull request closed without merging**